src/pages/workspace/WorkspaceAvatar.tsx

```diff
@@ -1,31 +1,27 @@
 import React from 'react';
-import type {OnyxEntry} from 'react-native-onyx';
-import {withOnyx} from 'react-native-onyx';
 import AttachmentModal from '@components/AttachmentModal';
+import useOnyx from '@hooks/useOnyx';
+import usePolicy from '@hooks/usePolicy';
 import Navigation from '@libs/Navigation/Navigation';
 import type {PlatformStackScreenProps} from '@libs/Navigation/PlatformStackNavigation/types';
 import type {AuthScreensParamList} from '@libs/Navigation/types';
-import * as ReportUtils from '@libs/ReportUtils';
-import * as UserUtils from '@libs/UserUtils';
+import {getDefaultWorkspaceAvatar} from '@libs/ReportUtils';
+import {getFullSizeAvatar} from '@libs/UserUtils';
 import ONYXKEYS from '@src/ONYXKEYS';
 import type SCREENS from '@src/SCREENS';
-import type {Policy} from '@src/types/onyx';
 
-type WorkspaceAvatarOnyxProps = {
-    policy: OnyxEntry<Policy>;
-    isLoadingApp: OnyxEntry<boolean>;
-};
+type WorkspaceAvatarProps = PlatformStackScreenProps<AuthScreensParamList, typeof SCREENS.WORKSPACE_AVATAR>;
 
-type WorkspaceAvatarProps = WorkspaceAvatarOnyxProps & PlatformStackScreenProps<AuthScreensParamList, typeof SCREENS.WORKSPACE_AVATAR>;
-
-function WorkspaceAvatar({policy, isLoadingApp = true}: WorkspaceAvatarProps) {
-    const avatarURL = (policy?.avatarURL ?? '') ? (policy?.avatarURL ?? '') : ReportUtils.getDefaultWorkspaceAvatar(policy?.name ?? '');
+function WorkspaceAvatar({route}: WorkspaceAvatarProps) {
+    const policy = usePolicy(route?.params?.policyID);
+    const [isLoadingApp = true] = useOnyx(ONYXKEYS.IS_LOADING_APP, {canBeMissing: true});
```

Comment:
Even though it wasn't there in the `withOnyx` usage, I strongly suspect we want to prevent the ephemeral loading state from being read from disk:

```suggestion
    const [isLoadingApp = true] = useOnyx(ONYXKEYS.IS_LOADING_APP, {canBeMissing: true, initWithStoredValues: false});
```

Reply:
That seems equivalent to having the default value be `false`, because the default value of `undefined` is falsy. So overall I think it would be better to do:

```ts
const [isLoadingApp = false] = useOnyx(ONYXKEYS.IS_LOADING_APP, { canBeMissing: true, initWithStoredValues: false });
```

because then the local `isLoadingApp` variable has type `boolean` rather than `boolean | undefined`

Reply:
> I don't see a case where isLoading could be undefined except initWithStoredValues=false

I don't understand this - as far as I can tell, it's only ever not-undefined if you've at some point logged in as a copilot to the account. Perhaps I'm missing something though